src/platform.ts

```diff
@@ -134,100 +129,61 @@ export class PlatformInformation {
+        }
+        else if (platform === 'darwin') {
+            return new PlatformInformation(platform, await PlatformInformation.GetUnixArchitecture());
+        }
+        else if (platform === 'linux') {
+            const [isMusl, architecture, distribution] = await Promise.all([
+                PlatformInformation.GetIsMusl(),
+                PlatformInformation.GetUnixArchitecture(),
+                LinuxDistribution.GetCurrent()
+            ]);
+            return new PlatformInformation(isMusl ? 'linux-musl' : platform, architecture, distribution);
         }
 
-        const platformData: [string, LinuxDistribution] = await Promise.all([architecturePromise, distributionPromise]);
-
-        return new PlatformInformation(platform, platformData[0], platformData[1]);
+        throw new Error(`Unsupported platform: ${platform}`);
     }
 
-    private static async GetWindowsArchitecture(): Promise<string> {
-        return new Promise<string>((resolve, reject) => {
-            if (process.env.PROCESSOR_ARCHITECTURE === 'x86' && process.env.PROCESSOR_ARCHITEW6432 === undefined) {
-                resolve('x86');
-            }
-            else if (process.env.PROCESSOR_ARCHITECTURE === 'ARM64' && process.env.PROCESSOR_ARCHITEW6432 === undefined) {
-                resolve('arm64');
-            }
-            else {
-                resolve('x86_64');
-            }
-        });
+    private static GetWindowsArchitecture(): string {
+        if (process.env.PROCESSOR_ARCHITECTURE === 'x86' && process.env.PROCESSOR_ARCHITEW6432 === undefined) {
+            return 'x86';
+        }
+        else if (process.env.PROCESSOR_ARCHITECTURE === 'ARM64' && process.env.PROCESSOR_ARCHITEW6432 === undefined) {
+            return 'arm64';
+        }
+        else {
+            return 'x86_64';
+        }
     }
 
-    private static async GetUnixArchitecture(): Promise<string | undefined> {
-        let architecture = await util.execChildProcess('uname -m');
-
-        if (architecture) {
-            architecture = architecture.trim();
-
-            switch (architecture) {
-                case "aarch64":
-                    return "arm64";
-                default:
-                    return architecture;
-            }
+    private static async GetUnixArchitecture(): Promise<string> {
+        const architecture = (await util.execChildProcess('uname -m')).trim();
+        if (architecture === "aarch64") {
+            return "arm64";
         }
-
-        return undefined;
+        return architecture;
     }
 
-    private static async GetIsMusl() {
-        let output;
+    private static async GetIsMusl(): Promise<boolean> {
         try {
-            output = await util.execChildProcess('ldd --version');
+            const output = await util.execChildProcess('ldd --version');
+            return output.includes('musl');
         } catch (err) {
-            output = err.message;
+            return false;
```

Comment:
The goal was to replicate this bit of code from the [dotnet-install.sh](https://github.com/dotnet/install-scripts/blob/3c6cc06b5054af7afcc5cbb4e659b551cc03e236/src/dotnet-install.sh#L187-L189) which treats stderr the same as stdout.

```sh
is_musl_based_distro() {
  (ldd --version 2>&1 || true) | grep -q musl
}
```

```suggestion
            return err.message.includes('musl');
```

Reply:
Thanks for the context, took suggestion & updated it to work with err typed as unknown.